src/AspNet.Security.OAuth.DingTalk/DingTalkAuthenticationHandler.cs

```diff
@@ -0,0 +1,290 @@
+        }
+
+        // See https://developers.dingtalk.com/document/app/obtain-orgapp-token for details.
+        protected override async Task<OAuthTokenResponse> ExchangeCodeAsync([NotNull] OAuthCodeExchangeContext context)
+        {
+            var address = QueryHelpers.AddQueryString(Options.TokenEndpoint, new Dictionary<string, string?>
+            {
+                ["appkey"] = Options.ClientId,
+                ["appsecret"] = Options.ClientSecret
+            });
+            using var response = await Backchannel.GetAsync(address);
+            if (!response.IsSuccessStatusCode)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following payload: {Headers} {Body}.",
+                                /* Status: */ response.StatusCode,
+                                /* Headers: */ response.Headers.ToString(),
+                                /* Body: */ await response.Content.ReadAsStringAsync());
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            var content = await response.Content.ReadAsStringAsync();
+            using var payload = JsonDocument.Parse(content);
+
+            var status = payload.RootElement.GetProperty("errcode").GetInt32();
+            if (status != 0)
+            {
+                Logger.LogError("An error occurred while retrieving the user profile: the remote server " +
+                                "returned a {Status} response with the following message: {Message}.",
+                                /* Status: */ status,
+                                /* Message: */  payload.RootElement.GetString("errmsg"));
+
+                throw new HttpRequestException("An error occurred while retrieving user information.");
+            }
+
+            return OAuthTokenResponse.Success(payload);
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(
+            [NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties,
+            [NotNull] OAuthTokenResponse tokens)
+        {
+            var code = Request.Query["code"];
+            var userInfoTmp = await GetUserInfoTmpAsync(code);
+            var userIdentifier = await GetUserIdentifierAsync(userInfoTmp.unionid!, tokens.AccessToken);
+
+            // See https://developers.dingtalk.com/document/app/query-user-details for details.
+            var address = QueryHelpers.AddQueryString(Options.UserInformationEndpoint, "access_token", tokens.AccessToken);
+            using var requestContent = new ReadOnlyMemoryContent(await CopyPayloadAsync(new Dictionary<string, Microsoft.Extensions.Primitives.StringValues>
+            {
+                ["userid"] = userIdentifier.userid,
+                ["language"] = Options.Language
+            }));
```

Comment:
Use  `JsonSerializer.SerializeToUtf8Bytes(new { userid = userIdentifier.userid, language = Options.Language })`  instead?

Reply:
> Use `JsonSerializer.SerializeToUtf8Bytes(new { userid = userIdentifier.userid, language = Options.Language })` instead?

Yes, it should output the same result and make the related codes simpler. 
But an anonymous object is an instance of reference type, since the size of payload object is always very short, I think that defining a private struct(do not use ValueTuple) is more suitable.